docs/source/DEVELOPER.rst

```diff
@@ -144,3 +144,29 @@ The Deepcell Kiosk uses Google Kubernetes Engine to requisition resources on Goo
+---------------------------------------------------------
+
+The DeepCell Kiosk comes with a utility for benchmarking the scalability and performance of a deep learning workflow. To run your own benchmarking, please read below.
+
+1. If you don't already have a cloud storage bucket devoted exclusively to DeepCell Kiosk benchmarking runs, you should create one now. It's fine to reuse this bucket for multiple benchmarking runs, but don't use it as the storage bucket for normal DeepCell Kiosk cluster operations.
+
+2. Then, checkout the ``benchmarks`` branch of the ``kiosk-console`` repository. This branch is almost identical to the version 1.2.0 release of the ``kiosk-console`` repo, except for minor configuration differences. There are three variable settings worth noting in the benchmarking pod's YAML file, ``conf/helmfile.d/0410.benchmarking.yaml``, each with a corresponding setup action for the user to perform before benchmarking:
+
+    - In the benchmarking pod's YAML file, ``conf/helmfile.d/0410.benchmarking.yaml``, the ``MODEL`` environmental variable has been set to the exact model name and version we used in our benchmarking, ``NuclearSegmentation:2``. The benchmarking cluster, when you create it, will look for the model files corresponding to this name and version in the ``models/`` folder of your benchmarking bucket. To recreate our benchmarking data, please copy the ``models/NuclearSegmentation`` folder in the Van Valen Lab's `kiosk-benchmarking bucket <https://console.cloud.google.com/storage/browser/kiosk-benchmarking>`_ to the same location in your bucket.
+    - Also in the benchmarking pod's YAML file is an environmental variable, ``FILE``, which is set to ``zip100.zip``. The benchmarking pod will look for a file by this name in the bucket you specified above at the location ``uploads/[file_name]``. We have a file named ``zip100.zip`` located in our ``kiosk-bencharking`` bucket at ``sample_data/zip100.zip``. This file consists of 100 microscopy images, which we used as the basis for all our benchmarking runs. To recreate our benchmarking data, please copy ``sample_data/zip100.zip`` from the Van Valen Lab's `kiosk-benchmarking bucket <https://console.cloud.google.com/storage/browser/kiosk-benchmarking>`_ to ``uploads/zip100.zip`` in your bucket.
+    - A final variable in the benchmarking pod's YAML file is ``COUNT``, which determines how many times the ``zip100.zip`` file will be submitted to the cluster for processing. This controls the total number of images the cluster will process over the course of the benchmarking run. For example, if you want to do a 10,000-image benchmarking run, you would set ``COUNT`` TO 100, since 100*100 = 10,000. Benchmarking data was presented in the paper for 10,000-image, 100,000-image, and 1,000,000-image runs. Accordingly, depending on which subset of the paper's benchmarking data the user wishes to recreate, ``COUNT`` should be set to either 100, 1,000, or 10,000.
```

Comment:
The intro section already says there are three variables in the benchmarking helmfile that need to be changed, we don't really need to explain in each item that the varaible is in the benchmarking helmfile. It might be simpler to just reformat as a bullet list:

> There are three variable settings worth noting in the benchmarking pod's YAML file, ``conf/helmfile.d/0410.benchmarking.yaml``, each with a corresponding setup action for the user to perform before benchmarking:
> - `MODEL` : the model name and version of the model in the bucket, e.g. `NuclearSegmentation:2`.
> - `FILE`: The datafile that will be processed by a deep learning job
> - `COUNT`: The number of jobs to create

Reply:
Yeah, the number of references to the YAML file can definitely be cut down. You think I should remove all the explanation on choosing values and copying files?

Reply:
I think just tightening up the explanations will work. Maybe a template like:

> - `COUNT`: The number of jobs used for benchmarking. Values of 100, 1000,  10000 were used in the paper.